File: python/world_2/examples/6_condicoes_(if..elif)/condicoes aninhadas/ex037_binario_octal_hexadecimal/ex037.js
( () => {

    const divHTML = document.querySelector( '[data-result]' );
    let message;

    const buttonSendInfos = document.querySelector( '[data-send-button]' );
    buttonSendInfos.addEventListener( 'click' , function () {

        let inputNumber = Number(document.getElementById( 'input1' ).value);
        let inputOptions = Number(document.getElementById( 'input2' ).value);
        
        let binary = inputNumber.toString(2)
        let octal = inputNumber.toString(8)
        let hexadecimal = inputNumber.toString(16)

        if ( validInputs( inputNumber , inputOptions ) ) {
            switch(inputOptions) {

                case 1: 
                    message = `O valor de ${inputNumber} para binário é ${binary}`;
                    showInfo( message , 'true' );
                    break;

                case 2: 
                    message = `O valor de ${inputNumber} para octal é ${octal}`;
                    showInfo( message , 'true' );
                    break;

                case 3: 
                    message = `O valor de ${inputNumber} para hexadecimal é ${hexadecimal}`;
                    showInfo( message , 'true' );
                    break;

                case 4: 
                    message = `O valor de ${inputNumber} para binário é ${binary}, para octal é ${octal} e para hexadecimal é ${hexadecimal}`;
                    showInfo( message , 'true' );
                    break;

            }
        }

        
        
    });

    const validInputs = ( number1 , number2 ) => {

        if ( number1 == '' || number2 == '') {
            message = 'Não pode haver campos vazios!';
            showInfo( message , 'error' );
            return true;
        }

        if ( isNaN( ( number1 ) ) || isNaN( ( number2 ) ) ) {
            message = 'Informe um valor válido!';
            showInfo( message , 'error' );
            return true;
        }

        if ( number2 < 1 || number2 > 4) {
            message = 'Digite uma opção válida!';
            showInfo( message , 'error' );
            return true;
        }

    return validInputs;

}

    const showInfo = ( message , colorIdentificationMessage ) => {
    divHTML.innerHTML = ''
    divHTML.innerHTML = /* html */ `<div class="answer ${colorIdentificationMessage}">${ message }</div>`;
}

}) ();

File: python/world_2/examples/6_condicoes_(if..elif)/condicoes aninhadas/ex037_binario_octal_hexadecimal/ex037.py
# Escreva um programa que leia um numero inteiro qualquer e peça para o usuário escolher qual será a base de conversão:

# 1 - para binário
# 2 - para octal
# 3 - para hexadecimal

number = int(input('Informe um número: '))
opcao = int(input('Você quer convertê-lo para: \n1-Binário; \n2-Octal; \n3-Hexadecimal: \n'))
conversao = 0
    

if opcao == 1:
    conversao = bin(number)
    print(f'binario: {conversao}')

elif opcao == 2:
    conversao = oct(number)
    print(f'octal: {conversao}')
    

elif opcao == 3:
    conversao = hex(number)
    print(f'hexadecimal: {conversao}')

else:
    print('Digite uma opção válida!')
